src/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -696,10 +696,27 @@ public override Task SendResponseAsync(HttpStatusCode statusCode = HttpStatusCod
             return SendResponseHeadersAsync(streamId, endStream : isFinal, statusCode, isTrailingHeader : false, endHeaders : endHeaders, headers);
         }
 
+        public override Task SendResponseHeadersAsync(HttpStatusCode statusCode = HttpStatusCode.OK, IList<HttpHeaderData> headers = null, int requestId = 0)
+        {
+            int streamId = requestId == 0 ? _lastStreamId : requestId;
+            return SendResponseHeadersAsync(streamId, endStream: false, statusCode, isTrailingHeader: false, endHeaders: true, headers);
+        }
+
         public override Task SendResponseBodyAsync(byte[] body, bool isFinal = true, int requestId = 0)
         {
             int streamId = requestId == 0 ? _lastStreamId : requestId;
             return SendResponseBodyAsync(streamId, body, isFinal);
         }
+
+        public override async Task WaitForCancellationAsync()
```

Comment:
Should this take a streamId and verify it?

Also, it's a little loose that we just discard any non-RstStream frame. At the very least it seems like we should verify that the discarded frames are DATA frames on the specified streamId. Better might be to make this an option on the method, e.g. "ignoreIncomingData" or whatever, so that a test can be written that expects RstStream (or more generally) cancellation immediately, by setting this false, and a different test can allow for eventual cancellation.

Reply:
> Should this take a streamId and verify it?

Sure.

> At the very least it seems like we should verify that the discarded frames are DATA frames on the specified streamId

Ok.